mathesar/database/collections.py

```diff
@@ -0,0 +1,40 @@
+from sqlalchemy import Column, Integer, String, Table
+
+from mathesar.database.applications import DBApplication
+from mathesar.database.base import ID, db_name, engine, metadata
+
+
+class DBCollection(Table):
+    DEFAULT_COLUMNS = [
+        Column(ID, Integer, primary_key=True),
+    ]
+
+    def insert_rows(self, rows):
+        with engine.begin() as connection:
+            result = connection.execute(self.insert(), rows)
+            return result
+
+    @classmethod
+    def create_from_columns(cls, name, column_names):
+        """
+        This method creates a Postgres table corresponding to the collection.
+        """
+        db_application = DBApplication(name)
+        db_application.create()
+        columns = cls.DEFAULT_COLUMNS + [
+            Column(column_name, String) for column_name in column_names
+        ]
+        db_collection = cls(
+            db_name(name),
+            metadata,
+            *columns,
+            schema=db_application.schema,
+        )
+        metadata.create_all(engine, tables=[db_collection])
```

Comment:
There's also a `Table.create` method available that automatically restricts to just the table in question.